nixos/modules/services/web-apps/c2FmZQ-server.nix

```diff
@@ -0,0 +1,138 @@
+{ lib, pkgs, config, ... }:
+
+let
+  inherit (lib) mkEnableOption mkPackageOption mkOption types;
+
+  cfg = config.services.c2FmZQ-server;
+
+  argsFormat = with lib; {
+    type = with types; let
+      valueType = nullOr (oneOf [
+        bool int str
+      ]) // {
+        description = "c2FmZQ-server params";
+      };
+    in valueType;
+    generate = name: value:
+      let mkParam = k: v:
+            if v == null then []
+            else if isBool v then optional v ("--"+k)
+            else [("--"+k) v];
+      in escapeShellArgs (builtins.concatLists (mapAttrsToList mkParam value));
```

Comment:
This exists :grin: 

```suggestion
    generate = lib.cli.toGNUCommandLineShell { };
```

Reply:
Thanks, this makes the code quite a bit cleaner.